client/client_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2020 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository at
+// https://github.com/direct-state-transfer/dst-go
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package client_test
+
+import (
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/direct-state-transfer/dst-go/blockchain/ethereum/ethereumtest"
+	"github.com/direct-state-transfer/dst-go/client"
+	"github.com/direct-state-transfer/dst-go/comm/tcp"
+	"github.com/direct-state-transfer/dst-go/node/nodetest"
+)
+
+func Test_Integ_NewEthereumPaymentClient(t *testing.T) {
+	// NOTE : This integration setup requires manaual setup of a ganache-cli node.
+	// Run the below command in a terminal before running this test.
+	//
+	// ganache-cli \
+	// --account="0x7d51a817ee07c3f28581c47a5072142193337fdca4d7911e58c5af2d03895d1a,100000000000000000000000" \
+	// --account="0x6aeeb7f09e757baa9d3935a042c3d0d46a2eda19e9b676283dce4eaf32e29dc9,100000000000000000000000"
+	//
+	// Deploy asset holder and adjudicator contracts.A quick hacky solution is to clone the project at
+	// github.com/perun-network/perun-eth-demo, build it and run the below cmd:
```

Comment:
```suggestion
	// github.com/perun-network/perun-eth-demo, checkout out dccb8d7, build it and run the below cmd:
```  
we changed it to `Alice`, so better fix it to a specific version.

Reply:
In the comments, I have added the git revision of the code to be used.

Reply:
I have added setup functions to deploy the contracts. You can find it in [`integration_setup_test.go`](https://github.com/direct-state-transfer/dst-go/pull/50/commits/4babc40b93da3e6c2d770d3085e51154916a8f89#diff-9620ed9af0d0854b0730c37b41c8f6de). Hence I have also removed the reference to `perun-eth-demo`.

Reply:
Yes this is much better.